pkg/ui/bucket.go

```diff
@@ -48,30 +48,44 @@ func NewBucketUI(logger log.Logger, label, externalPrefix, prefixHeader, uiPrefi
 }
 
 // Register registers http routes for bucket UI.
-func (b *Bucket) Register(r *route.Router, registerNewUI bool, ins extpromhttp.InstrumentationMiddleware) {
+func (b *Bucket) Register(r *route.Router, ins extpromhttp.InstrumentationMiddleware) {
+	/*	redirectPath := "/blocks"
+		if len(b.uiPrefix) > 0 {
+			redirectPath = b.uiPrefix
+		}
+		r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+			http.Redirect(w, r, path.Join(GetWebPrefix(b.logger, b.externalPrefix, b.prefixHeader, r), redirectPath), http.StatusFound)
+		})
+		// Redirect the original React UI's path (under "/new") to its new path at the root.
+		r.Get("/new/*path", func(w http.ResponseWriter, r *http.Request) {
+			p := route.Param(r.Context(), "path")
+			http.Redirect(w, r, path.Join(GetWebPrefix(b.logger, b.externalPrefix, b.prefixHeader, r), strings.TrimPrefix(p, "/new"))+"?"+r.URL.RawQuery, http.StatusFound)
+		})
+
+		// here we have two routes that serve the same document. It's because it depends where do we come from.
+		// If we are coming from the new UI, it will use the first route.
+		// If we are coming from the old UI, it will use the second route.
+		r.Get(path.Join("/classic", b.uiPrefix), instrf("bucket", ins, b.bucket))
+		r.WithPrefix(b.uiPrefix).Get("/classic", instrf("bucket", ins, b.bucket))
```

Comment:
`panic: a handle is already registered for path '/classic'`

I got this panic when running Block Viewer locally. If the `uiPrefix` is empty string, this will try to register `/classic` again, which causes a panic.

Reply:
Ah, I guess the `uiPrefix` is also a block viewer specific artifact as well, stemming from the classic UI.

Reply:
yeah ... I guess I have to put a condition about the uiPrefix like I did a bit above